neuralforecast/tsdataset.py

```diff
@@ -370,12 +408,180 @@ def __init__(
         self.target_col = target_col
         self.min_size = min_size
 
-# %% ../nbs/tsdataset.ipynb 11
+# %% ../nbs/tsdataset.ipynb 10
+class LocalFilesTimeSeriesDataset(BaseTimeSeriesDataset):
+
+    def __init__(
+        self,
+        files_ds: List[str],
+        temporal_cols,
+        id_col: str,
+        time_col: str,
+        target_col: str,
+        last_times,
+        indices,
+        max_size: int,
+        min_size: int,
+        y_idx: int,
+        static=None,
+        static_cols=None,
+        sorted=False,
+    ):
+        super().__init__(
+            temporal_cols=temporal_cols,
+            max_size=max_size,
+            min_size=min_size,
+            y_idx=y_idx,
+            static=static,
+            static_cols=static_cols,
+            sorted=sorted,
+        )
+        self.files_ds = files_ds
+        self.id_col = id_col
+        self.time_col = time_col
+        self.target_col = target_col
+        # array with the last time for each timeseries
+        self.last_times = last_times
+        self.indices = indices
+        self.n_groups = len(files_ds)
+
+    def __getitem__(self, idx):
+        if not isinstance(idx, int):
+            raise ValueError(f"idx must be int, got {type(idx)}")
+
+        temporal_cols = self.temporal_cols.copy()
+        data = pd.read_parquet(self.files_ds[idx], columns=temporal_cols).to_numpy()
```

Comment:
Sorry, it seems we do need this.
```suggestion
        data = pd.read_parquet(self.files_ds[idx], columns=temporal_cols.tolist()).to_numpy()
```

Reply:
Sounds good - when I was testing it seemed to work with the pd.Index but showed up as a typing error. This fixes that.